### ISSUE TITLE
Convert nodes to float

### ISSUE THREAD

**Issue by anntzer** [dhermes/bezier#145]
Minor usability request: currently, the Curve constructor (didn't check for Surface) happily accepts integer node arrays as input, but trying to do any operation on a curve defined with integer nodes throws an exception:
```
In [2]: bezier.Curve.from_nodes(np.array([[1, 2, 1], [3, 4, 5]]))                                                   
Out[2]: <Curve (degree=2, dimension=2)>

In [3]: bezier.Curve.from_nodes(np.array([[1, 2, 1], [3, 4, 5]])).evaluate(.5)                                      
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-369133655e72> in <module>
----> 1 bezier.Curve.from_nodes(np.array([[1, 2, 1], [3, 4, 5]])).evaluate(.5)

~/.local/lib/python3.7/site-packages/bezier/curve.py in evaluate(self, s)
    206         """
    207         return _curve_helpers.evaluate_multi(
--> 208             self._nodes, np.asfortranarray([s])
    209         )
    210 

src/bezier/_speedup.pyx in bezier._speedup.evaluate_multi()

ValueError: Buffer dtype mismatch, expected 'double' but got 'long'
```
Perhaps the constructor could just cast the array to doubles (or at the very least the constructor should be the one throwing the error).

**Comment by dhermes** [comment]
Thanks for bringing this to my attention. I **just now** realized I have hardcoded the scalar type as `double` in `_speedup.pyx` (and in my `.f90` files) but the Python code makes no such assumptions / has no such requirements. I will have to update the `Curve` and `Surface` constructors to check `nodes.dtype == np.float64`.

----

What's worse, the pure-Python implementation (i.e. the stuff that doesn't come from `_speedup.pyx`) will do just fine on this input. For example, if I install without speedups:

```
python3.7 -m virtualenv venv
BEZIER_NO_EXTENSIONS=True venv/bin/python -m pip install bezier==0.9.0 --no-binary=bezier
venv/bin/python -m pip install ipython
```

then running your example (via `venv/bin/ipython`):

```python
In [1]: import numpy as np

In [2]: import bezier

In [3]: bezier._HAS_SPEEDUP
Out[3]: False

In [4]: bezier.Curve.from_nodes(np.array([[1, 2, 1], [3, 4, 5]])).evaluate(.5)
Out[4]:
array([[1.5],
       [4. ]])
```

**Comment by dhermes** [comment]
Fixed in 68f7dc7c1f26bb678d09b4221fd917531fb79860
